src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs

```diff
@@ -134,12 +136,27 @@ protected static async Task<string> RenderRazorPageAsync(Compilation compilation
 
         // Create ViewContext.
         var appBuilder = WebApplication.CreateBuilder();
-        appBuilder.Services.AddMvc().AddApplicationPart(assembly);
+        appBuilder.Services.AddMvc().ConfigureApplicationPartManager(manager =>
+        {
+            var partFactory = new ConsolidatedAssemblyApplicationPartFactory();
+            foreach (var applicationPart in partFactory.GetApplicationParts(assembly))
+            {
+                manager.ApplicationParts.Add(applicationPart);
+            }
+        });
         var app = appBuilder.Build();
         var httpContext = new DefaultHttpContext
         {
             RequestServices = app.Services
         };
+        var requestFeature = new HttpRequestFeature
+        {
+            Method = HttpMethods.Get,
+            Protocol = HttpProtocol.Http2,
+            Scheme = "http"
+        };
+        requestFeature.Headers.Host = "localhost";
+        httpContext.Features.Set<IHttpRequestFeature>(requestFeature);
```

Comment:
Same as in https://github.com/dotnet/razor/pull/8721: small improvement of the test infra to allow rendering also Blazor components.